boot/bootstate20.go

```diff
@@ -137,8 +145,9 @@ func (ks20 *bootState20Kernel) revisions() (curSnap, trySnap snap.PlaceInfo, try
 	}
 
 	tryKernel, err := ks20.ebl.TryKernel()
+	// if err is ErrNoTryKernelRef, then we will just return nil as the trySnap
 	if err != nil && err != bootloader.ErrNoTryKernelRef {
-		return nil, nil, "", fmt.Errorf("cannot identify try kernel snap with bootloader %s: %v", ks20.ebl.Name(), err)
+		return bootSn, nil, "", trySnapError(fmt.Sprintf("cannot identify try kernel snap with bootloader %s: %v", ks20.ebl.Name(), err))
```

Comment:
wondering if newTrySnapError(fmt, a...) would be slightly nicer?

Reply:
yes, I can do that easily